libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1541,18 +1543,21 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 )
 {
 	FunctionType const& funType = dynamic_cast<FunctionType const&>(type(_functionCall.expression()));
-	solAssert(
-		funType.takesArbitraryParameters() ||
-		_arguments.size() == funType.parameterTypes().size(), ""
-	);
+	solAssert(!funType.takesArbitraryParameters(), "");
+	solAssert(_arguments.size() == funType.parameterTypes().size(), "");
 	solUnimplementedAssert(!funType.bound(), "");
 	FunctionType::Kind const funKind = funType.kind();
 
-	solAssert(funKind != FunctionType::Kind::BareStaticCall || m_context.evmVersion().hasStaticCall(), "");
-	solAssert(funKind != FunctionType::Kind::BareCallCode, "Callcode has been removed.");
+	solAssert(funKind != FunctionType::Kind::BareCall, "Should use appendBareCall");
+	solAssert(funKind != FunctionType::Kind::BareDelegateCall, "Should use appendBareCall");
+	solAssert(funKind != FunctionType::Kind::BareStaticCall, "Should use appendBareCall");
+	solAssert(funKind != FunctionType::Kind::BareCallCode, "Should use appendBareCall");
 
-	bool const isDelegateCall = funKind == FunctionType::Kind::BareDelegateCall || funKind == FunctionType::Kind::DelegateCall;
-	bool const useStaticCall = funKind == FunctionType::Kind::BareStaticCall || (funType.stateMutability() <= StateMutability::View && m_context.evmVersion().hasStaticCall());
+	solUnimplementedAssert(funKind != FunctionType::Kind::RIPEMD160, "");
+	solUnimplementedAssert(funKind != FunctionType::Kind::ECRecover, "");
```

Comment:
What's actually handled here in the end :-)? Maybe
```
solAssert(funKind == FunctionType::Kind::External || funKind == FunctionType::Kind::DelegateCall, "");
```
?
E.g. is ``FunctionType::Kind::SHA256`` handled here or is it just missing its assertion? Will ``RIPEMD160`` and ``ECRecover`` likely use this or a separate function as well?

Reply:
sorry, this part is still work in progress.

Reply:
The reason is that I changed my mind later on: I think we should use `appendExternalFunctionCall` only for user-defined functions. Specialized functions like sha256, ripemd and so on should be implemented somewhere else in order to keep the complexity of this template bounded.

Reply:
If we don't implement those here, we should check that everything works well with `try / catch` though.